src/liboslnoise/simplexnoise.cpp

```diff
@@ -315,56 +354,86 @@ OSL_HOSTDEVICE float simplexnoise2 (float x, float y, int seed,
-
-    float t = (float)(i+j+k)*G3; 
-    float X0 = i-t; // Unskew the cell origin back to (x,y,z) space
-    float Y0 = j-t;
-    float Z0 = k-t;
-    float x0 = x-X0; // The x,y,z distances from the cell origin
-    float y0 = y-Y0;
-    float z0 = z-Z0;
+    float s  = (x + y + z) * F3;  // Very nice and simple skew factor for 3D
+    float xs = x + s;
+    float ys = y + s;
+    float zs = z + s;
+    int i    = OIIO::ifloor(xs);
+    int j    = OIIO::ifloor(ys);
+    int k    = OIIO::ifloor(zs);
+
+    float t  = (float)(i + j + k) * G3;
+    float X0 = i - t;  // Unskew the cell origin back to (x,y,z) space
+    float Y0 = j - t;
+    float Z0 = k - t;
+    float x0 = x - X0;  // The x,y,z distances from the cell origin
+    float y0 = y - Y0;
+    float z0 = z - Z0;
 
     // For the 3D case, the simplex shape is a slightly irregular tetrahedron.
     // Determine which simplex we are in.
-    int i1, j1, k1; // Offsets for second corner of simplex in (i,j,k) coords
-    int i2, j2, k2; // Offsets for third corner of simplex in (i,j,k) coords
+    int i1, j1, k1;  // Offsets for second corner of simplex in (i,j,k) coords
+    int i2, j2, k2;  // Offsets for third corner of simplex in (i,j,k) coords
 
 #if 1
     // TODO: This code would benefit from a backport from the GLSL version!
     // (no it can't... see note below)
-    if (x0>=y0) {
-        if (y0>=z0) {
-            i1=1; j1=0; k1=0; i2=1; j2=1; k2=0;  /* X Y Z order */
-        } else if (x0>=z0) {
-            i1=1; j1=0; k1=0; i2=1; j2=0; k2=1;  /* X Z Y order */
+    if (x0 >= y0) {
+        if (y0 >= z0) {
+            i1 = 1;
+            j1 = 0;
+            k1 = 0;
+            i2 = 1;
+            j2 = 1;
+            k2 = 0; /* X Y Z order */
+        } else if (x0 >= z0) {
+            i1 = 1;
+            j1 = 0;
+            k1 = 0;
+            i2 = 1;
+            j2 = 0;
+            k2 = 1; /* X Z Y order */
         } else {
-            i1=0; j1=0; k1=1; i2=1; j2=0; k2=1;  /* Z X Y order */
+            i1 = 0;
+            j1 = 0;
+            k1 = 1;
+            i2 = 1;
+            j2 = 0;
+            k2 = 1; /* Z X Y order */
         }
-    } else { // x0<y0
-        if (y0<z0) {
-            i1=0; j1=0; k1=1; i2=0; j2=1; k2=1;  /* Z Y X order */
-        } else if (x0<z0) {
-            i1=0; j1=1; k1=0; i2=0; j2=1; k2=1;  /* Y Z X order */
+    } else {  // x0<y0
+        if (y0 < z0) {
+            i1 = 0;
+            j1 = 0;
+            k1 = 1;
+            i2 = 0;
+            j2 = 1;
+            k2 = 1; /* Z Y X order */
+        } else if (x0 < z0) {
+            i1 = 0;
+            j1 = 1;
+            k1 = 0;
+            i2 = 0;
+            j2 = 1;
+            k2 = 1; /* Y Z X order */
         } else {
-            i1=0; j1=1; k1=0; i2=1; j2=1; k2=0;  /* Y X Z order */
+            i1 = 0;
+            j1 = 1;
+            k1 = 0;
+            i2 = 1;
+            j2 = 1;
+            k2 = 0; /* Y X Z order */
         }
```

Comment:
This was more readable prior to clang-format.

Reply:
fixing